06x-new-kafka-roller.md

```diff
@@ -0,0 +1,350 @@
+# Kafka Roller 2.0
+
+## Current situation
+
+The Kafka Roller is an internal Cluster Operator component that's responsible for coordinating the rolling restart or reconfiguration of Kafka pods when:
+- non-dynamic reconfigurations needs to be applied
+- update in Kafka CRD is detected
+- a certificate is renewed
+- pods have been manually annotated by the user for controlled restarts
+- pod is stuck and is out of date
+- Kafka broker is unresponsive to Kafka Admin connections
+
+A pod is considered stuck if it is in one of following states:
+- `CrashLoopBackOff`
+- `ImagePullBackOff`
+- `ContainerCreating`
+- `Pending` and `Unschedulable`
+
+### Known Issues
+
+The existing KafkaRoller suffers from the following shortcomings:
+- Although it is safe and straightforward to restart one broker at a time, this process is slow in large clusters ([related issue](https://github.com/strimzi/strimzi-kafka-operator/issues/8547)).
```

Comment:
I think historically, we had way more users asking us to _slow down_ the rolling update in various issues by introducing a cool down interval between the restarts. Maybe this should be included here as well as an optional (default to 0) delay.